plugins/meta/route-override/route-override.go

```diff
@@ -0,0 +1,400 @@
+				link, _ := netlink.LinkByName(netif.Name)
+				routes, _ := netlink.RouteList(link, netlink.FAMILY_ALL)
+				for _, nlroute := range routes {
+					if nlroute.Dst == nil {
+						err = netlink.RouteDel(&nlroute)
+					}
+				}
+			}
+		}
+	}
+
+	return err
+}
+
+func deleteRoute(route *types.Route, res *current.Result) error {
+	var err error
+	// fallback to eth0 if there is no interface in result
+	if res.Interfaces == nil {
+		link, _ := netlink.LinkByName("eth0")
+		routes, _ := netlink.RouteList(link, netlink.FAMILY_ALL)
+		for _, nlroute := range routes {
+			if nlroute.Dst != nil &&
+				nlroute.Dst.IP.Equal(route.Dst.IP) &&
+				nlroute.Dst.Mask.String() == route.Dst.Mask.String() {
+				err = netlink.RouteDel(&nlroute)
+			}
+		}
+	} else {
+		for _, netif := range res.Interfaces {
+			if netif.Sandbox != "" {
+				link, _ := netlink.LinkByName(netif.Name)
+				routes, _ := netlink.RouteList(link, netlink.FAMILY_ALL)
+				for _, nlroute := range routes {
+					if nlroute.Dst != nil &&
+						nlroute.Dst.IP.Equal(route.Dst.IP) &&
+						nlroute.Dst.Mask.String() == route.Dst.Mask.String() {
+						err = netlink.RouteDel(&nlroute)
+					}
+				}
+			}
+		}
+	}
+
+	return err
+}
+
+func addRoute(dev netlink.Link, route *types.Route) error {
+	return netlink.RouteAdd(&netlink.Route{
+		LinkIndex: dev.Attrs().Index,
+		Scope:     netlink.SCOPE_UNIVERSE,
+		Dst:       &route.Dst,
+		Gw:        route.GW,
+	})
+}
+
+func processRoutes(netnsname string, conf *RouteOverrideConfig) (*current.Result, error) {
+	netns, err := ns.GetNS(netnsname)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open netns %q: %v", netns, err)
+	}
+	defer netns.Close()
+
+	res, err := current.NewResultFromResult(conf.PrevResult)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert result to current version: %v", err)
+	}
+
+	if conf.FlushGateway {
+		// add "0.0.0.0/0" into delRoute to remove it from routing table/result
+		_, gwRoute, _ := net.ParseCIDR("0.0.0.0/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+		_, gwRoute, _ = net.ParseCIDR("::/0")
+		conf.DelRoutes = append(conf.DelRoutes, &types.Route{Dst: *gwRoute})
+
+		// delete given gateway address
+		for _, ips := range res.IPs {
+			if ips.Version == "6" {
+				ips.Gateway = net.IPv6zero
+			} else {
+				ips.Gateway = net.IPv4zero
+			}
+		}
+	}
+
+	newRoutes := []*types.Route{}
+	err = netns.Do(func(_ ns.NetNS) error {
+		// Flush route if required
+		if !conf.FlushRoutes {
+		NEXT:
+			for _, route := range res.Routes {
+				for _, delroute := range conf.DelRoutes {
+					if route.Dst.IP.Equal(delroute.Dst.IP) &&
+						bytes.Equal(route.Dst.Mask, delroute.Dst.Mask) {
+						err = deleteRoute(delroute, res)
+						if err != nil {
+							fmt.Fprintf(os.Stderr, "failed to delte route %v: %v", delroute, err)
+						}
+						continue NEXT
+					}
+
+				}
+				newRoutes = append(newRoutes, route)
+			}
+		} else {
+			deleteAllRoutes(res)
```

Comment:
Do not ignore errors returned there - they should be at least logged as a warning.